### ISSUE TITLE
Bug in LogitLens Tutorial Plotting

### ISSUE THREAD

**Issue by oliveradk** [ndif-team/nnsight#462]
The plotly cell used in the logit lens tutorial https://nnsight.net/notebooks/tutorials/logit_lens/ has a bug: 

If there are duplicate `input_words`, plotly removes the duplicates, which offsets the columns and (leads to confusing results)

adding the following snip-it solves the problem (though possible there are less verbose methods)

```
def make_unique_with_invisible_chars(strings):
    seen = {}
    result = []
    for string in strings:
        if string in seen:
            seen[string] += 1
            modified_string = string + '\u200B' * seen[string]
            result.append(modified_string)
        else:
            seen[string] = 0
            result.append(string)
    return result
input_words = make_unique_with_invisible_chars(input_words)
```
(original cell)
```python
import plotly.express as px
import plotly.io as pio

if is_colab:
    pio.renderers.default = "colab"
else:
    pio.renderers.default = "plotly_mimetype+notebook_connected+notebook"

fig = px.imshow(
    max_probs.detach().cpu().numpy(),
    x=input_words,
    y=list(range(len(words))),
    color_continuous_scale=px.colors.diverging.RdYlBu_r,
    color_continuous_midpoint=0.50,
    text_auto=True,
    labels=dict(x="Input Tokens", y="Layers", color="Probability")
)

fig.update_layout(
    title='Logit Lens Visualization',
    xaxis_tickangle=0
)

fig.update_traces(text=words, texttemplate="%{text}")
fig.show()
```

**Comment by JordanMcCann** [comment]
@oliveradk's diagnosis is right. go.Heatmap and px.imshow treat the x list as categorical, so duplicate tokens - a repeated "the" or "cat" - land on the same column, and everything after shifts.

The zero-width-space trick works, but you can sidestep the dedup entirely. Give the heatmap numeric x positions and move the tokens to tick labels. No invisible characters, labels stay exact, one column per token:

  ```python
  xpos = list(range(len(input_words)))
  fig = go.Figure(go.Heatmap(z=probs, x=xpos, y=layers, colorscale="..."))
  fig.update_xaxes(tickmode="array", tickvals=xpos, ticktext=input_words)
  ```

  Same idea for px.imshow: pass `x=xpos`, then `fig.update_xaxes(tickvals=xpos, ticktext=input_words)`.

Root cause is Plotly mapping duplicate string categories onto the same x-position, so columns after a repeat shift under their labels. I checked on plotly 6.7 that the positional-x version keeps every column with the exact labels. Happy to PR it against the logit lens tutorial.

**Comment by davidbau** [comment]
@Belfki, Adam ***@***.***> should we revise the tutorial to
use the modern logit lens widget instead of plotly?

**Comment by AdamBelfki3** [comment]
Absolutely! That's a great point, I'll update this tutorial and perhaps the activation patching one too.
